Linker error with Crystal v1.16.0 on Windows for xml2.lib

I am getting the following linker error with Crystal v1.16.0 on Windows when compiling a project that uses XML:

```batch
crystal build -o C:\Temp\Project\main.cr --verbose --release --static
"C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.41.34120\bin\Hostx64\x64\cl.exe" /nologo _main.o3.obj /FeC:\Temp\Project\bin\main.exe /link "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.41.34120\atlmfc\lib\x64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.41.34120\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64" /DEBUG:FULL /PDBALTPATH:%_PDB% /INCREMENTAL:NO /STACK:0x800000 /LIBPATH:C:\Crystal\v1.16.0\lib /ENTRY:wmainCRTStartup C:\Crystal\v1.16.0\lib\mpir.lib C:\Crystal\v1.16.0\lib\xml2.lib C:\Crystal\v1.16.0\lib\z.lib "C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\user32.lib" "C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\crypt32.lib" C:\Crystal\v1.16.0\lib\crypto.lib C:\Crystal\v1.16.0\lib\ssl.lib C:\Crystal\v1.16.0\lib\pcre2-8.lib C:\Crystal\v1.16.0\lib\gc.lib "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.41.34120\lib\x64\libcmt.lib" "C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\ntdll.lib" C:\Crystal\v1.16.0\lib\iconv.lib "C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\mswsock.lib" "C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\advapi32.lib" "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.41.34120\lib\x64\libvcruntime.lib" "C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\shell32.lib" "C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\ole32.lib" "C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\ws2_32.lib" "C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\kernel32.lib" "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.41.34120\lib\x64\legacy_stdio_definitions.lib" "C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x64\DbgHelp.lib" "C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\ucrt\x64\libucrt.lib"
LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library
LINK : warning LNK4217: symbol 'ispunct' defined in 'libucrt.lib(_ctype.obj)' is imported by 'pcre2-8.lib(pcre2_context.obj)' in function 'pcre2_set_glob_escape_8'
LINK : warning LNK4217: symbol 'free' defined in 'libucrt.lib(free.obj)' is imported by 'pcre2-8.lib(pcre2_context.obj)' in function 'default_free'
LINK : warning LNK4217: symbol 'malloc' defined in 'libucrt.lib(malloc.obj)' is imported by 'pcre2-8.lib(pcre2_context.obj)' in function 'default_malloc'
xml2.lib(dict.obj) : error LNK2019: unresolved external symbol BCryptGenRandom referenced in function xmlInitRandom
C:\Temp\Project\bin\main.exe : fatal error LNK1120: 1 unresolved externals
Error target main failed to compile:
Error: execution of command failed with exit status 2
```

Is the bundled `xml2.lib` in the Windows 1.16.0 release the correct version?

Note that the same project compiles without issue with Crystal v1.15.1.

--- a/src/xml/libxml2.cr
+++ b/src/xml/libxml2.cr
@@ -12,6 +12,9 @@ require "./save_options"
 @[Link("xml2", pkg_config: "libxml-2.0")]
 {% if compare_versions(Crystal::VERSION, "1.11.0-dev") >= 0 %}
   @[Link(dll: "libxml2.dll")]
+  {% if flag?("win32") %}
+    @[Link("BCrypt")]
+  {% end %}
 {% end %}
 lib LibXML
   alias Int = LibC::Int

--- a/src/xml/libxml2.cr
+++ b/src/xml/libxml2.cr
@@ -13,7 +13,7 @@ require "./save_options"
 {% if compare_versions(Crystal::VERSION, "1.11.0-dev") >= 0 %}
   @[Link(dll: "libxml2.dll")]
   {% if flag?("win32") %}
-    @[Link("BCrypt")]
+    @[Link("bcrypt")]
   {% end %}
 {% end %}
 lib LibXML

